src/components/progress/progress.ts

```diff
@@ -0,0 +1,28 @@
+import { Component, Input, ViewEncapsulation, HostBinding } from '@angular/core';
+
+@Component({
+  selector: 'ion-progress',
+  templateUrl : './progress.html',
+  encapsulation: ViewEncapsulation.None
+})
+export class Progress {
+
+  constructor( ) { }
+
+  /**
+   * @input {'determinate' | 'indeterminate' | 'buffer' | 'query'} The current mode of the progress indicator
+   */
+  @Input()
+  @HostBinding('class')
```

Comment:
According to the angular docs its best to use `Hostbinding` in contrast with host. But if you would like me to use `host` instead, I have no problem in doing so.